File: Python3/201 - 400/206 Reverse Linked List.py
# 52ms, 99.75%
class Solution:
    def reverseList(self, head):
        if head and head.next:
            last, trav = head, head.next
            while trav:
                last.next, trav.next, head = trav.next, head, trav
                trav = last.next
        return head
    

File: 201 - 400/206 Reverse Linked List.cc
// 9ms, 13.51%
#include <bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode *reverseList(ListNode *head) {
        if (!head) return head;
        ListNode *pre = head, *nxt = pre->next, *tmp = pre;
        while (nxt) {
            head = nxt;
            pre->next = nxt->next;
            nxt->next = tmp;
            nxt = pre->next;
            tmp = head;
        }
        return head;
    }
};
